src/core/metadata/qgslayermetadataformatter.cpp

```diff
@@ -0,0 +1,217 @@
+  return myMetadata;
+}
+
+QString QgsLayerMetadataFormatter::contactsSectionHtml( const QgsLayerMetadata &metadata ) const
+{
+  QString myMetadata = QStringLiteral( "<table width=\"100%\" class=\"tabular-view\">\n" );
+  myMetadata += "<tr><th>" % QObject::tr( "ID" ) % "</th><th>" % QObject::tr( "Name" ) % "</th><th>" % QObject::tr( "Position" ) % "</th><th>" % QObject::tr( "Organization" ) % "</th><th>" % QObject::tr( "Role" ) % "</th><th>" % QObject::tr( "Email" ) % "</th><th>" % QObject::tr( "Voice" ) % "</th><th>" % QObject::tr( "Fax" ) % "</th><th>" % QObject::tr( "Addresses" ) % "</th></tr>\n";
+
+  int i = 1;
+  const QList<QgsLayerMetadata::Contact> &contacts = metadata.contacts();
+  for ( const QgsLayerMetadata::Contact &contact : contacts )
+  {
+    QString rowClass( "" );
+    if ( i % 2 )
+      rowClass = QString( "class=\"odd-row\"" );
+    myMetadata += "<tr " % rowClass % "><td>" % QString::number( i ) % "</td><td>" % contact.name % "</td><td>" % contact.position % "</td><td>" % contact.organization % "</td><td>" % contact.role % "</td><td>" % contact.email % "</td><td>" % contact.voice % "</td><td>" % contact.fax % "</td><td>";
+    bool notFirstRow = false;
+    for ( const QgsLayerMetadata::Address &oneAddress : contact.addresses )
+    {
+      if ( notFirstRow )
+      {
+        myMetadata += QLatin1String( "<br />\n" );
+      }
+      if ( ! oneAddress.type.isEmpty() )
+      {
+        myMetadata += oneAddress.type % QLatin1String( "<br />" );
+      }
+      if ( ! oneAddress.address.isEmpty() )
+      {
+        myMetadata += oneAddress.address % QLatin1String( "<br />" );
+      }
+      if ( ! oneAddress.postalCode.isEmpty() )
+      {
+        myMetadata += oneAddress.postalCode % QLatin1String( "<br />" );
+      }
+      if ( ! oneAddress.city.isEmpty() )
+      {
+        myMetadata += oneAddress.city % QLatin1String( "<br />" );
+      }
+      if ( ! oneAddress.administrativeArea.isEmpty() )
+      {
+        myMetadata += oneAddress.administrativeArea % QLatin1String( "<br />" );
+      }
+      if ( ! oneAddress.country.isEmpty() )
+      {
+        myMetadata += oneAddress.country;
+      }
+      notFirstRow = true;
+    }
+    myMetadata += "</td></tr>\n";
+    i++;
+  }
+  myMetadata += QLatin1String( "</table>\n" );
+  return myMetadata;
+}
+
+QString QgsLayerMetadataFormatter::extentSectionHtml( const QgsLayerMetadata &metadata ) const
+{
+  QString myMetadata = QStringLiteral( "<table class=\"list-view\">\n" );
+  myMetadata += QStringLiteral( "<tr><td class=\"highlight\">" ) % QObject::tr( "CRS" ) % QLatin1String( "</td><td>" ) % metadata.crs().authid() % QLatin1String( "</td></tr>\n" );
+
+  const QgsLayerMetadata::Extent extent = metadata.extent();
+  myMetadata += QStringLiteral( "<tr><td class=\"highlight\">" ) % QObject::tr( "Spatial Extent" ) % QLatin1String( "</td><td>" );
+  QList< QgsLayerMetadata::SpatialExtent > spatialExtents = extent.spatialExtents();
```

Comment:
This will need to be `const QList< .... > spatialExtents = etc`, otherwise the for loop will detach it.